tools/luh2/luh2mod.py

```diff
@@ -0,0 +1,258 @@
+
+    return(input_dataset)
+
+# Updating datasets to work with xESMF
+def PrepDataset_ESMF(input_dataset,dsflag,dstype):
+
+    if (dsflag):
+        if(dstype == "LUH2"):
+            print("PrepDataset: LUH2")
+            input_dataset = BoundsVariableFixLUH2(input_dataset)
+        elif(dstype == "Surface"):
+            print("PrepDataset: SurfData")
+            input_dataset = DimensionFixSurfData(input_dataset)
+        print("data set updated for xESMF\n")
+
+    return(input_dataset)
+
+# Create the necessary variable "lat_b" and "lon_b" for xESMF conservative regridding
+# Each lat/lon boundary array is a 2D array corresponding to the bounds of each
+# coordinate position (e.g. lat_boundary would be 90.0 and 89.75 for lat coordinate
+# of 89.875).
+def BoundsVariableFixLUH2(input_dataset):
+
+    # Create lat and lon bounds as a single dimension array out of the LUH2 two dimensional_bounds array.
+    # Future todo: is it possible to have xESMF recognize and use the original 2D array?
+    input_dataset["lat_b"] = np.insert(input_dataset.lat_bounds[:,1].data,0,input_dataset.lat_bounds[0,0].data)
+    input_dataset["lon_b"] = np.insert(input_dataset.lon_bounds[:,1].data,0,input_dataset.lon_bounds[0,0].data)
+
+    # Drop the old boundary names to avoid confusion
+    input_dataset = input_dataset.drop(labels=['lat_bounds','lon_bounds'])
+
+    print("LUH2 dataset lat/lon boundary variables formatted and added as new variable for xESMF")
+
+    return(input_dataset)
+
+# The user will need to use a surface data set to regrid from, but the surface datasets
+# need to have their dimensions renamed to something recognizable by xESMF
+def DimensionFixSurfData(input_dataset):
+
+    # Rename the surface dataset dimensions to something recognizable by xESMF.
+    # input_dataset = surfdataset.rename_dims(dims_dict={'lsmlat':'latitude','lsmlon':'longitude'})
+    input_dataset = input_dataset.rename_dims(dims_dict={'lsmlat':'lat','lsmlon':'lon'})
+
+    # Populate the new surface dataset with the actual lat/lon values
+    # input_dataset['longitude'] = input_dataset.LONGXY.isel(latitude=0)
+    # input_dataset['latitude']  = input_dataset.LATIXY.isel(longitude=0)
+    input_dataset['lon'] = input_dataset.LONGXY.isel(lat=0)
+    input_dataset['lat'] = input_dataset.LATIXY.isel(lon=0)
+
+    print("Surface dataset dimensions renamed for xESMF")
+
+    return(input_dataset)
+
+# def SetMask(input_dataset, masking_dataset):
+
+#     # check what sort of inputdata is being provided; surface dataset or luh2
+#     # LUH2 data will need to be masked based on the variable input to mask
+#     dsflag,dstype = CheckDataset(input_dataset)
+#     if (dsflag):
+#         if(dstype == "LUH2"):
+#             SetMaskLUH2(input_dataset) # temporary
+#         elif(dstype == "Surface"):
+#             SetMaskSurfData(input_dataset)
+#         print("mask added")
+#
+#     return(input_dataset)
```

Comment:
@glemieux 
reminder to delete obsolete commented out code here and elsewhere.

Reply:
completed with e44bc90ffb895100051b7ed6305a4fa3625c5db9